src/Api/ApiRequest.php

```diff
@@ -62,27 +65,25 @@ public function post($url, $params)
      * API Request Get Method.
      *
      * @param string $url
-     * @param array $params
+     * @param string |null $params
      * @return array |null
      * @throws \Exception
      */
-    public function get($url, $params = [])
+    public function get($url, $params = null)
     {
-
         try {
             $response = Request::get(
                 $this->getBaseUrl() . $url,
                 $this->getHeaders(),
                 $params
             );
 
-            #TODO: Add validation http codes and handlers
-            //$this->isValidResponse($response);
+            $this->isValidResponse($response);
 
             return json_decode($response->raw_body);
         } catch (\Exception $exception) {
             $logs = new SaferPayLog();
-            $logs->message = $exception->getResponse() ? $exception->getResponse()->getBody()->getContents() : 'missing response';
+            $logs->message = $exception->getMessage() ? : "missing response";
```

Comment:
is this valid syntax? Shouldnt it be  $logs->message = $exception->getMessage() ?: "missing response";